utils/analysis/factory.go

```diff
@@ -104,6 +104,9 @@ func ValidateMetric(metric v1alpha1.Metric) error {
 	if metric.Provider.Job != nil {
 		numProviders++
 	}
+	if metric.Provider.Wavefront != nil {
```

Comment:
Can you add a quick unit test to https://github.com/argoproj/argo-rollouts/blob/master/utils/analysis/factory_test.go#L77 for this case?